snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -96,10 +97,14 @@ def _get_source_handler(self, properties):
         # the default '.'
         source_handler = None
         if properties['source']:
+            if 'source' in self._part_property_grammar:
+                source = self._grammar_processor.get_source()
```

Comment:
Rather than saving off the grammar before defaults are applied, could you not just call this function to see if it returns a source? If not, use the default.

Reply:
can you expand or maybe just fix it? :slightly_smiling_face:

Reply:
Yep, on it.

Reply:
Fixed.